### PR TITLE
HTTP routes to serve files with correct content type and etag headers

### LINKED ISSUE #541: Select current view for blob using custom middleware
this would mean we don't need to symlink between blobs and their current view

### PR DESCRIPTION
**Pull request by adzialocha**
This PR switches from using symlinks and the `ServeDir` struct from `tower` to HTTP routes which resolve the blobs manually and respond with the correct content types.

Notable changes:

* Files are now directly stored into the `/blobs` folder, without sub-directories using their document view id
* HTTP requests to `/blobs/:documentId` respond with the latest blob
* HTTP requests to `/blobs/:documentId/:viewId` respond with the blob at that view id (this makes `/blobs/documents/...` obsolete
* Responses contain `Content-Type` header set by the blob's self-declared MIME type
* Responses contain `ETag` header
* Requests with a `If-None-Match` header might be responded with a 304 "Not Modified" in case the entity tags matched
* Added `update_blob` test method
* Changed `add_blob` test method to slice up blob data into "pieces" by given "max piece length" variable
* Renamed test client method to reflect that it is not anymore for GraphQL only
* Changed all `fs` methods to async `tokio::fs`

Closes #541 and #501

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header

### CODE REVIEW

File: aquadoggo/src/materializer/tasks/blob.rs [hunk 1]
```diff
@@ -75,38 +73,36 @@ pub async fn blob_task(context: Context, input: TaskInput) -> TaskResult<TaskInp
...
             None => return Err(TaskError::Critical("No base path configured".to_string())),
         };
-
-        let blob_dir = base_path.join(blob_document.id().as_str());
-
-        fs::create_dir_all(&blob_dir).map_err(|err| TaskError::Critical(err.to_string()))?;
-        let blob_view_path = blob_dir.join(blob_document.view_id().to_string());
+        // @TODO: Why is this required here? Shouldn't the folder already exist at this point? (if
+        // I remove this writing a file fails ..)
+        fs::create_dir_all(base_path).unwrap();
```
Comment (adzialocha) [inline aquadoggo/src/materializer/tasks/blob.rs:85]: This `create_dir_all` is a little bit of a mystery for me. If I remove it, creating a file fails, even though the folder should have been already created when the whole node got initialised?

File: aquadoggo/src/test_utils/node.rs [hunk 2]
```diff
@@ -161,7 +161,7 @@ pub async fn add_document(
         // We only want to issue dependency tasks.
         let dependency_tasks = tasks
             .iter()
-            .filter(|task| task.worker_name() == "depenedency");
+            .filter(|task| task.worker_name() == "dependency");
```
Comment (adzialocha) [inline aquadoggo/src/test_utils/node.rs:164]: 😅

Comment (sandreae) [reply aquadoggo/src/test_utils/node.rs:164]: Guess we never really wanted to issue dependency tasks either 😅

File: aquadoggo/src/materializer/tasks/blob.rs (hunk 1 above)
Comment (adzialocha) [reply aquadoggo/src/materializer/tasks/blob.rs:85]: I would like to explore this a bit more when we have the configuration interface (where `base_path` might change into something outside of `/tmp`)

Comment (adzialocha) [reply aquadoggo/src/materializer/tasks/blob.rs]: Change this now to an async tokio version of this method .. that seems to do the trick?

File: aquadoggo/src/materializer/tasks/blob.rs [hunk 3]
```diff
@@ -75,38 +71,33 @@ pub async fn blob_task(context: Context, input: TaskInput) -> TaskResult<TaskInp
...
+            ))
+        })?;
+
+        file.write_all(&blob_data).await.map_err(|err| {
+            TaskError::Critical(format!(
+                "Could not write blob file @ {}: {}",
+                blob_view_path.display(),
+                err
+            ))
+        })?;
```
Comment (adzialocha) [inline aquadoggo/src/materializer/tasks/blob.rs:100]: I've changed this to use the `tokio::fs` methods, this assures that writing to files is done in a separate thread and doesn't block the main thread :+1:

**Review by sandreae — CHANGES_REQUESTED** [review]
Looks great to me, very exciting to have the last few blob features coming in.

**Review by sandreae — APPROVED** [review]

**Pull request merged**